### PR TITLE
[MIRPrinter] Don't print line break when there is no instructions (NFC)

### PR DESCRIPTION
**Pull request by dianqk**
Per #80143, we can remove the extra line break when there is no instruction.

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-scalar-float-sop2.mir [hunk 1]
```diff
@@ -14,8 +14,8 @@ body:             |
     ; GFX1150-NEXT: {{  $}}
     ; GFX1150-NEXT: [[COPY:%[0-9]+]]:sreg_32 = COPY $sgpr0
     ; GFX1150-NEXT: [[COPY1:%[0-9]+]]:sreg_32 = COPY $sgpr1
-    ; GFX1150-NEXT: %2:sreg_32 = nofpexcept S_ADD_F32 [[COPY]], [[COPY1]], implicit $mode
-    ; GFX1150-NEXT: $sgpr0 = COPY %2
+    ; GFX1150-NEXT: [[S_ADD_F32_:%[0-9]+]]:sreg_32 = nofpexcept S_ADD_F32 [[COPY]], [[COPY1]], implicit $mode
```
Comment (jayfoad) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-scalar-float-sop2.mir:17]: You could avoid some churn here by first regenerating the checks in this file and committing it as obvious, and then rebasing the current patch on top of that. Same for any other files which have unrelated changes like this.

File: llvm/lib/CodeGen/MIRPrinter.cpp [hunk 2]
```diff
@@ -728,7 +728,7 @@ void MIPrinter::print(const MachineBasicBlock &MBB) {
     HasLineAttributes = true;
   }
 
-  if (HasLineAttributes)
+  if (HasLineAttributes && !MBB.instrs().empty())
```
Comment (jayfoad) [inline llvm/lib/CodeGen/MIRPrinter.cpp:731]: Makes sense to me: you only need the blank line separator if there are some successors/liveins *and* some instructions.

**Review by jayfoad — APPROVED** [review]

File: llvm/lib/CodeGen/MIRPrinter.cpp (hunk 2 above)
Comment (DianQK) [reply llvm/lib/CodeGen/MIRPrinter.cpp:731]: > Makes sense to me: you only need the blank line separator if there are some successors/liveins _and_ some instructions.

`HasLineAttributes` should already contain successors and liveins.

File: llvm/test/CodeGen/AArch64/GlobalISel/arm64-irtranslator-switch.ll [hunk 3]
```diff
@@ -136,9 +136,8 @@ define i32 @test_cfg_remap_multiple_preds(i32 %in) {
   ; CHECK-NEXT: {{  $}}
   ; CHECK-NEXT: bb.2.odd:
   ; CHECK-NEXT:   successors:
-  ; CHECK: {{  $}}
```
Comment (DianQK) [inline llvm/test/CodeGen/AArch64/GlobalISel/arm64-irtranslator-switch.ll:139]: It looks like we've had this problem before.

File: llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-scalar-float-sop2.mir (hunk 1 above)
Comment (DianQK) [reply llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-scalar-float-sop2.mir]: I cleaned it up, and it looks much better now.

File: llvm/lib/CodeGen/MIRPrinter.cpp (hunk 2 above)
Comment (DianQK) [reply llvm/lib/CodeGen/MIRPrinter.cpp]: > Can simplify to MBB.empty?

Yes! Done.

**Comment by DianQK** [comment]
Thanks for the review. I'll add `NFC` to the commit message, which should make it less painful for others to view the history.

**Comment by DianQK** [comment]
@jayfoad @arsenm Do you have any suggestions for this PR? I would like to revert it until the end of the backport of LLVM 18. Otherwise, this could make backport difficult for the next few months.

**Comment by jayfoad** [comment]
> @jayfoad @arsenm Do you have any suggestions for this PR? I would like to revert it until the end of the backport of LLVM 18. Otherwise, this could make backport difficult for the next few months.

Is there a specific backport you're worried about? Personally I would not revert. For most targets it only affects a few tests, and even if a backport does cause conflicts it should be easy to fix them manually.

**Comment by DianQK** [comment]
> > @jayfoad @arsenm Do you have any suggestions for this PR? I would like to revert it until the end of the backport of LLVM 18. Otherwise, this could make backport difficult for the next few months.
> 
> Is there a specific backport you're worried about? Personally I would not revert. For most targets it only affects a few tests, and even if a backport does cause conflicts it should be easy to fix them manually.

This happens to affect my #79820 PR. 😅
Yes. It's easy, so let's look at it this way for a moment.
